engine/src/main/java/org/terasology/rendering/ShaderManagerLwjgl.java

```diff
@@ -180,6 +180,17 @@ private GLSLMaterial prepareAndStoreShaderProgramInstance(String title, ShaderPa
         return material;
     }
 
+    @Override
+    public void loadShader(ResourceUrn shaderName, ShaderParameters shaderParameters) {
```

Comment:
I think we should eliminate the code duplication between this method and prepareAndStoreShaderProgramInstance(String, ShaderParameters).

I think this can be achieved by placing the common code in a separate private method "_loadShader" or "internalLoadShader". Then, loadShader only checks that the namespace in the urn is NOT "engine:".  Meanwhile prepareAndStoreShaderProgramInstance just adds the "engine:" part to the input string. Both methods than call "_loadShader" to do the actual shader loading.